### ISSUE TITLE
[Potential Bug/Explanation Req.] complete_checkout will always create address duplicate if user is specified.

### ISSUE THREAD

**Issue by Awruk8** [saleor/saleor#8721]
### Discussed in https://github.com/saleor/saleor/discussions/8701

<div type='discussions-op-text'>

<sup>Originally posted by **Awruk8** December  2, 2021</sup>
Hello Saleor team,

Before I make it into an Issue, 
I wanted to make sure that I am not missing something,
and what I presume is a bug is after all "a feature".

```
class Address(models.Model):
    ....
    def get_copy(self):
        """Return a new instance of the same address."""
        return Address.objects.create(**self.as_data())
```

```
def _process_shipping_data_for_order
    ....
    if checkout_info.user and shipping_address:
        ...
        if checkout_info.user.addresses.filter(pk=shipping_address.pk).exists():
            shipping_address = shipping_address.get_copy()
```
From the above I deduce that whenever an Order is about to be created.  
If the `checkout_info` contains the `user` and `shipping_address` (and usually, I think, it does),  
in case if the used address is associated with the user, a copy will be created.
The copy is another record in the database.
In a solution I want to built I'm attempting to make multiple orders from an internal user (simplified: a bot).
So there would be multiple orders created for the same user and the same address. 
That will result in database address table growing with each order created.

**I want to know what is the intention in creating a copy (in database) for the shipping address?**
Because it seems it is put there deliberately. 
And since I could not figure out why is that on my own having a quick look at the code,
perhaps a good idea would be to add some in-code comment explaining it.

I can of course bypass the check, by having the checkout_info not contain the user information.
However, in my solution there would be potentially a high number of `bots` each associated with different supplier etc.
So it might be nice to have the internal Orders related to the corresponding user.
</div>

**Comment by Awruk8** [comment]
Received no feedback in 'Discussions' tab, so I've decided to propagate it into an Issue hoping for a response.

**Comment by timuric** [comment]
@Awruk8 we have addressed this in 3.1 there would no more than X amount of orders per email
